### PR TITLE
feature: Added app_state support for applications #5068

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fixes: #5068

Application support needs some major loving and a rewrite, for now this will provide a status + prev_status to be useful for alerting.

### CODE REVIEW

**Comment by laf** [comment]
Rebased

File: includes/polling/functions.inc.php [hunk 1]
```diff
@@ -564,3 +564,37 @@ function get_device_oid_limit($device)
         return 10;
     }
 }
+
+/**
+ * Update the application status and output in the database.
+ *
+ * @param array $device
```
Comment (murrant) [inline includes/polling/functions.inc.php]: Why do we need $device?

Comment (laf) [inline includes/polling/functions.inc.php]: Good spot, I've made more use of it by adding it to the dbUpdate call where it should have been included anyway.

File: includes/polling/functions.inc.php [hunk 2]
```diff
@@ -596,5 +596,5 @@ function update_applications($device, $app, $response)
     if ($data['app_state'] != $app['app_state']) {
         $data['app_state_prev'] = $app['app_state'];
     }
-    dbUpdate($data, 'applications', '`app_id` = ?', array($app['app_id']));
+    dbUpdate($data, 'applications', '`app_id` = ? AND `device_id` = ?', array($app['app_id'], $device['device_id']));
```
Comment (murrant) [inline includes/polling/functions.inc.php]: app_id is unique, why do we need to check device_id too?

Comment (laf) [inline includes/polling/functions.inc.php]: Being cautious?

**Pull request merged**